core/iwasm/aot/aot_runtime.c

```diff
@@ -1049,7 +1049,24 @@ memory_instantiate(AOTModuleInstance *module_inst, AOTModuleInstance *parent,
     return NULL;
 }
 
-static AOTMemoryInstance *
+AOTMemoryInstance *
+aot_lookup_memory(AOTModuleInstance *module_inst, char const *name)
+{
+#if WASM_ENABLE_MULTI_MEMORY != 0
+    uint32 i;
+    for (i = 0; i < module_inst->export_memory_count; i++)
+        if (!strcmp(module_inst->export_memories[i].name, name))
+            return module_inst->export_memories[i].memory;
+    return NULL;
+#else
+    (void)module_inst->export_memories;
+    if (!module_inst->memories)
+        return NULL;
+    return module_inst->memories[0];
```

Comment:
I believe that functions `lookup_xx()` are intended to focus on exported functions, global variables, tables, and memories, as Spec. required. However, lines 1062 to 1065 seem to return the default memory, which is causing confusion.

Reply:
Yes it is a bit confusing, but for this aot_lookup_memory() implementation, I directly mimicked wasm_lookup_memory() which @wenyongh originally implemented. Do you think that implementation needs to be fixed in some way also?

Reply:
Oh, I understand the issue now. WAMR currently doesn't support the export and import of memory and tables. That's why we implemented a temporary solution. We're also considering updating our approach to meet the specification requirements. It might be the right moment to correct this. 

However, it's still up to you to choose whether to include this implementation in the current pull request or not.

@wenyongh, what's your opinion?

Reply:
Agree, had better also modify wasm_lookup_memory (and wasm_lookup_table if needed). But since module_inst->export_memories isn't created, seems we should lookup the module's export table to check whether the memory is exported.

Reply:
I implemented exporting of memory and tables a while back. Once this PR lands, I also plan to do another PR to add support for memory imports, and hopefully tables after that.

Reply:
Excellent 🎆. I'm planning to work on implementing instantiation linking, which necessitates complete support for export and import functionality. Your assistance will help expedite this task.

Reply:
@lum1n0us So let's merge this PR first and wait for @bnason-nf's next PR?